ledger-app-fio/src/textUtils.c

```diff
@@ -0,0 +1,121 @@
+#include "textUtils.h"
+#include "hexUtils.h"
+
+#define WRITE_CHAR(ptr, end, c) \
+	{ \
+		ASSERT(ptr + 1 <= end); \
+		*ptr = (c); \
+		ptr++; \
+	}
+
+size_t str_formatFIOAmount(uint64_t amount, char* out, size_t outSize)
+{
+	ASSERT(outSize < BUFFER_SIZE_PARANOIA);
+
+	char scratchBuffer[30];
+	char* ptr = BEGIN(scratchBuffer);
+	char* end = END(scratchBuffer);
+
+	// We print in reverse
+
+	// decimal digits
+	for (int dec = 0; dec < 9; dec++) {
+		WRITE_CHAR(ptr, end, '0' + (amount % 10));
+		amount /= 10;
+	}
+	WRITE_CHAR(ptr, end, '.');
+	// We want at least one iteration
+	int place = 0;
+	do {
+		// thousands separator
+		if (place && (place % 3 == 0)) {
+			WRITE_CHAR(ptr, end, ',');
+		}
+		WRITE_CHAR(ptr, end, '0' + (amount % 10));
+		amount /= 10;
+		place++;
+	} while (amount > 0);
+
+	// Size without terminating character
+	STATIC_ASSERT(sizeof(ptr - scratchBuffer) == sizeof(size_t), "bad size_t size");
+	size_t rawSize = (size_t) (ptr - scratchBuffer);
+
+	const char *suffix = " FIO";
+	const size_t suffixLength = strlen(suffix);
+
+	if (rawSize + suffixLength + 1 > outSize) {
+		THROW(ERR_DATA_TOO_LARGE);
+	}
+
+	// Copy reversed & append terminator
+	for (size_t i = 0; i < rawSize; i++) {
+		out[i] = scratchBuffer[rawSize - 1 - i];
+	}
+	out[rawSize] = 0;
+
+	snprintf(out + rawSize, outSize - rawSize, "%s", suffix);
+	ASSERT(strlen(out) == rawSize + suffixLength);
+
+	return rawSize + suffixLength;
+}
+
+size_t str_formatUint64(uint64_t number, char* out, size_t outSize)
```

Comment:
Maybe we don't need this, but there is no harm keeping it here (if not used, it is cut away by the linker afaik).